project/frontend/src/containers/MySGonks/MySGonks.js

```diff
@@ -3,25 +3,106 @@ import classes from "./MySGonks.module.css";
 import Block from "../../components/UI/Block/Block";
 import LinkButton from "../../components/UI/LinkButton/LinkButton";
 import LongSGonksList from "../../components/SGonksLists/LongSGonksList/LongSGonksList";
+import ChartCard from "../../components/ChartCard/ChartCard";
 
 const MySGonks = (props) => {
+  const ONE_WEEK_MILLISECONDS = 7 * 24 * 60 * 60 * 1000;
+  const ONE_DAY_MILLISECONDS = 24 * 60 * 60 * 1000;
+  const INITIAL_WORTH = 500;
+
   const unsoldInvestments = !props.investments
     ? undefined
     : props.investments.filter(
-        (investment) => investment.dateSoldMilliSeconds == 0
+        (investment) => investment.dateSoldMilliSeconds === 0
       );
 
+  const getUsername = () => {
+    const rank = props.generalInfo.rank - 1
+    return props.rankings[rank].name
+  }
+
+  const getEarliestDate = (investments) => {
+    var currentEarliest = Infinity;
+    for (var i = 0; i < investments.length; i++) {
+      currentEarliest = Math.min(currentEarliest, investments[i].dateInvestedMilliSeconds);
+    }
+    return currentEarliest - ONE_WEEK_MILLISECONDS;
+  }
+
+  const getTitleChartRow = (investments) => {
+    var titleRow = ['x'];
+    for (var i = 0; i < investments.length; i++) {
+      var search = investments[i].searchItem;
+      titleRow.push(search);
+    }
+    return titleRow;
+  }
+
+  const populateInvestmentData = (investments, data, earliestDate, currentDate) => {
+    // add each investment's datapoints to correct chart indices
+    for (var i = 0; i < investments.length; i++) {
+      var firstDateWithData = investments[i].dateInvestedMilliSeconds - ONE_WEEK_MILLISECONDS;
+      var lastDateWithData = investments[i].dateSoldMilliseconds;
+      var rowCount = 1; // start at second row
+      var dataCount = 0;
+      var date = earliestDate;
+      var row;
+      while (date <= currentDate) {
+        if (date < firstDateWithData || date > lastDateWithData) {
+          row = data[rowCount];
```

Comment:
row = data[rowCount]; and           data[rowCount] = row;
Common to both if/else, can come out?